core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3608,25 +3633,29 @@ private SyncResult syncInodeMetadata(RpcContext rpcContext, LockedInodePath inod
         }
 
         // Iterate over Alluxio children and process persisted children.
-        for (Map.Entry<String, Inode> inodeEntry : inodeChildren.entrySet()) {
+        Stream<Map.Entry<String, Inode>> childStream = inodeChildren.entrySet().stream();
+        childStream.map(inodeEntry -> {
           if (!inodeEntry.getValue().isPersisted()) {
             // Ignore non-persisted inodes.
-            continue;
+            return SyncResult.defaults();
           }
 
           // Technically we don't need to lock here since inodePath is already write-locked. We can
           // improve this by implementing a way to traverse an inode path without locking.
           try (LockedInodePath descendant = inodePath.lockDescendant(
               inodePath.getUri().joinUnsafe(inodeEntry.getKey()), LockPattern.WRITE_EDGE)) {
             // Recursively sync children
-            if (syncDescendantType != DescendantType.ALL) {
-              syncDescendantType = DescendantType.NONE;
+            DescendantType type = syncDescendantType;
+            if (type != DescendantType.ALL) {
+              type = DescendantType.NONE;
             }
-            SyncResult syncResult =
-                syncInodeMetadata(rpcContext, descendant, syncDescendantType, statusCache);
-            pathsToLoad.addAll(syncResult.getPathsToLoad());
+            return syncInodeMetadata(rpcContext, descendant, type, statusCache);
+          } catch (InvalidPathException | FileDoesNotExistException | IOException
+              | AccessControlException e) {
+            LOG.error("Failed to process inode path: {}", inodeEntry.getValue().toString(), e);
```

Comment:
Is there more info we can add to this error message? For example, should this say "sync" instead of "process"?

Reply:
updated